src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,264 @@
+import {
+  HTMLAttributes,
+  MouseEvent,
+  ReactNode,
+  forwardRef,
+  useId,
+  useRef,
+  useState,
+} from "react";
+import * as RadixPopover from "@radix-ui/react-popover";
+import { ReactSortable } from "react-sortablejs";
+import { Badge, BadgeProps, Icon, Label } from "@/components";
+import { Error, FormElementContainer, FormRoot } from "../commonElement";
+import styled from "styled-components";
+import {
+  ComboboxGroup,
+  ComboboxItem,
+  ComboboxNoData,
+  ComboboxTrigger,
+  ComboboxContent,
+} from "../Combobox/ComboBoxElements";
+import { useCombobox } from "../Combobox/useCombobox";
+import { ComboboxProvider } from "../Combobox/ComboboxProvider";
+import { DismissibleBadge, NonDismissibleBadge } from "../Badge/Badge";
+
+interface Props extends Omit<HTMLAttributes<HTMLDivElement>, "onChange" | "dir"> {
+  label?: ReactNode;
+  children?: ReactNode;
+  error?: ReactNode;
+  disabled?: boolean;
+  defaultValue?: Array<string>;
+  onChange?: (value: Array<string>) => void;
+  name?: string;
+  required?: boolean;
+  isFormControl?: boolean;
+  value?: Array<string>;
+  dir?: "start" | "end";
+  orientation?: "horizontal" | "vertical";
+  onCreateOption?: (search: string) => void;
+  showCheck?: boolean;
+}
+
+export type MultiSelectProps = RadixPopover.PopoverProps & Props;
+
+const SelectPopoverRoot = styled(RadixPopover.Root)`
+  width: 100%;
+`;
+
+export const MultiSelect = ({
+  label,
+  children,
+  orientation,
+  dir,
+  disabled,
+  id,
+  error,
+  value: valueProp,
+  defaultValue,
+  onChange,
+  open: openProp,
+  defaultOpen,
+  onOpenChange: onOpenChangeProp,
+  name,
+  required,
+  isFormControl,
+  onCreateOption,
+  showCheck,
+  ...props
+}: MultiSelectProps) => {
+  const defaultId = useId();
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [open, setOpen] = useState(defaultOpen ?? openProp);
```

Comment:
i would avoid setting the internal open state as this can lead to inconsistencies such as a situation where the user `open` state is `false` but the select is open.

Reply:
This is added as in radix ui in order to have an option to not declare it externally and we can still have the functionality 
In this case the user can define the initial open state and not manage the open state externally and use the select like normally